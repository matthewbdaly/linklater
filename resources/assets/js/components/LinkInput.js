// @flow
import React, { Component } from 'react';
import ReactDOM from 'react-dom';

export default class LinkInput extends Component {
  constructor(props) {
    super(props);
    this.inputRef = React.createRef();
    this.submitLink = this.submitLink.bind(this);
  }
  submitLink(e: SyntheticEvent<HTMLButtonElement>) {
    e.preventDefault();
    let link = this.inputRef.current.value.trim();
    if (link) {
      this.props.storeLink(link);
      this.inputRef.current.value = '';
    }
  }
  render() {
    return (
      <div className="form-group">
        <label htmlFor="url" className="control-label">URL</label>
        <input type="url" name="url" placeholder="Page to save" className="form-control" required autoComplete="off" ref={this.inputRef}></input>
        <button type="submit" className="btn btn-primary" onClick={this.submitLink}>
          Submit
        </button>
      </div>
    );
  }
}
