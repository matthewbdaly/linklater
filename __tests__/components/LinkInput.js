import React from 'react';
import {findDOMNode} from 'react-dom';
import LinkInput from '../../resources/assets/js/components/LinkInput';
import renderer from 'react-test-renderer';
import { mount, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('LinkInput', () => {
  it('renders the link input', () => {
    const store = jest.fn();
    const tree = renderer
      .create(<LinkInput storeLink={store} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('handles submit', () => {
    const store = jest.fn();
    const wrapper = mount(<LinkInput storeLink={store} />);
    wrapper.find('button').simulate('click', {
      preventDefault: () => {
    }});
    expect(store).toBeCalled();
  });
});
