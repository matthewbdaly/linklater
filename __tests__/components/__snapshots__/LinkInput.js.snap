// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkInput renders the link input 1`] = `
<div
  className="form-group"
>
  <label
    className="control-label"
    htmlFor="url"
  >
    URL
  </label>
  <input
    autoComplete="off"
    className="form-control"
    name="url"
    placeholder="Page to save"
    required={true}
    type="url"
  />
  <button
    className="btn btn-primary"
    onClick={[Function]}
    type="submit"
  >
    Submit
  </button>
</div>
`;
